src/rp2_common/pico_stdio_usb/stdio_usb.c

```diff
@@ -25,32 +25,65 @@ static uint8_t stdio_usb_core_num;
 
 // when tinyusb_device is explicitly linked we do no background tud processing
 #if !LIB_TINYUSB_DEVICE
+// if this crit_sec is initialized, we are not in periodic timer mode, and must make sure
+// we don't either create multiple one shot timers, or miss creating one. this crit_sec
+// is used to protect the one_shot_timer_pending flag
+static critical_section_t one_shot_timer_crit_sec;
+static volatile bool one_shot_timer_pending;
 #ifdef PICO_STDIO_USB_LOW_PRIORITY_IRQ
 static_assert(PICO_STDIO_USB_LOW_PRIORITY_IRQ >= NUM_IRQS - NUM_USER_IRQS, "");
 #define low_priority_irq_num PICO_STDIO_USB_LOW_PRIORITY_IRQ
 #else
 static uint8_t low_priority_irq_num;
 #endif
 
+static int64_t timer_task(__unused alarm_id_t id, __unused void *user_data) {
+    assert(stdio_usb_core_num == get_core_num()); // if this fails, you have initialized stdio_usb on the wrong core
```

Comment:
> if this fails, you have initialized stdio_usb on the wrong core

Which core is this supposed to be? The one taking the IRQs for the default timer pool? Maybe it's worth making that constraint explicit in the doc comment for `stdio_usb_init`, especially if the outcome of this assertion is that you won't be getting any stdout.

Might it also be worth failing earlier, at init time, if the USB IRQ is not shared and the init is called on a different core than the default timer IRQ is serviced on?

Reply:
I have moved the assertion (it remains an assertion)... i added a return code to `stdio_init_all()` as it wasn't even previously passed thru.

Note, i also notice that `stdio_usb.c` won't even compile if `PICO_TIME_DEFAULT_ALARM_POOL_DISABLED=1` though no one has apparently needed this. It might be worth adding the ability to initialize `stdio_usb` with your own alarm pool instead, but that is for another day